### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by Navijava**

### CODE REVIEW

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 1]
```diff
@@ -3,45 +3,149 @@
...
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        } else if (index == 0) {
+            Node<T> nextNode = new Node<>(null, value, head);
+            head.prev = nextNode;
+            head = nextNode;
+        } else {
+            checkIndex(index);
```
Comment (pikador10) [inline src/main/java/core/basesyntax/MyLinkedList.java:45]: you can put checkIndex on the top of this method and check the index at once

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 2]
```diff
@@ -3,45 +3,149 @@
...
+            Node<T> targetNode = getNode(index);
+            Node<T> newnode = new Node<>(targetNode.prev, value, targetNode);
```
Comment (pikador10) [inline src/main/java/core/basesyntax/MyLinkedList.java:47]: better write newNode

**Review by pikador10 — APPROVED** [review]

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 3]
```diff
@@ -3,45 +3,149 @@
...
+            }
+        } else {
+            while (node != null) {
+                if (node.value.equals(object)) {
+                    unlink(node);
+                    return true;
+                }
+                node = node.next;
+            }
+        }
```
Comment (AlexRozv) [inline src/main/java/core/basesyntax/MyLinkedList.java:124]: we don't need this if-else construct and we can leave only one while loop if we change our if statement inside this while loop like this:

if (object == node.value || (object != null && object.equals(node.value)))

**Review by AlexRozv — CHANGES_REQUESTED** [review]

**Review by AlexRozv — APPROVED** [review]
well done ✅

**Pull request closed without merging**